src/zope/app/container/testing.py

```diff
@@ -1,7 +1,85 @@
-import os
-from zope.app.testing.functional import ZCMLLayer
+from zope.component.testing import PlacelessSetup
 
-AppContainerLayer = ZCMLLayer(
-    os.path.join(os.path.split(__file__)[0], 'ftesting.zcml'),
-    __name__, 'AppContainerLayer', allow_teardown=True)
+from zope.container.testing import setUp as cSetUp, tearDown as cTearDown
 
+try:
+    from zope.testing.cleanup import tearDown as tTearDown
+except ImportError: # pragma: no cover
+    def tTearDown():
+        pass
```

Comment:
`zope.testing` is not a normal dependency, only a test dependency, so it may not be available when this file is imported. Why you would do that I don't know.